noise/noise.go

```diff
@@ -89,6 +89,16 @@ func XKAndSecp256k1(config Config) (*Noise, error) {
 	return New(noise.HandshakeXK, config)
 }
 
+// GetEncNonce returns underlying encNonce.
+func (ns *Noise) GetEncNonce() uint64 {
+	return ns.encNonce
```

Comment:
I think we need to have locks here to avoid data races.

Reply:
Nice suggestion. 
What bad can happen if we read without lock counter which constantly increases? In the worst case, we will get not very accurate metrics which is not big deal on my opinion, but adding a lock for every session  will slowdown everyhting very much.